portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,28 +47,47 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     int commentLimit = Integer.parseInt(request.getParameter("limit"));
     String languageCode = request.getParameter("languageCode");
-    System.out.println(languageCode);
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<Entity> queryList = results.asList(FetchOptions.Builder.withLimit(commentLimit));
 
-    ArrayList<Comment> commentList = new ArrayList<Comment>();
-    for (Entity entity : queryList) {
-      long id = entity.getKey().getId();
-      String name = (String) entity.getProperty("name");
-      String comment = (String) entity.getProperty("comment");
-      long timestamp = (long) entity.getProperty("timestamp");
+    Translate translate = translate = TranslateOptions.getDefaultInstance().getService();
 
-      Comment commentObject = new Comment(id, name, comment, timestamp);
-      commentList.add(commentObject);
-    }
+    try {
+      ArrayList<Comment> commentList = new ArrayList<Comment>();
+      for (Entity entity : queryList) {
+        long id = entity.getKey().getId();
+        String name = (String) entity.getProperty("name");
+        String comment = (String) entity.getProperty("comment");
+        long timestamp = (long) entity.getProperty("timestamp");
+
+        // Do the translation.
+        if (!languageCode.equals("original")) {
+          Translation translation = translate.translate(comment, Translate.TranslateOption.targetLanguage(languageCode));
+          comment = translation.getTranslatedText();
+        }
+
+        Comment commentObject = new Comment(id, name, comment, timestamp);
+        commentList.add(commentObject);
+      }
     
-    Gson gson = new Gson();   
-    String json = gson.toJson(commentList);
-    response.setContentType("application/json; charset=utf-8");
-    response.getWriter().println(json);
+      Gson gson = new Gson();   
+      String json = gson.toJson(commentList);
+      response.setContentType("application/json; charset=utf-8");
+      response.getWriter().println(json);
+
+    } catch (TranslateException error) {
+      String errorMessage = "Google Translate API returned an error " + error.getMessage();
```

Comment:
usually error msg shown to the user could be something like "error translating", or "error translating message: <input msg>", you may not want to expose which api you use on server side to the user

Reply:
That makes sense. I added a commit which shortens the error message shown to the user as simply "Error translating message."